### PR TITLE
feat: [M3-7257] - Invalidate VPC-related queries after deleting a linode

### PR DESCRIPTION
**Pull request by coliu-akamai**
## Description 📝
- After deleting a linode, because we didn't invalidate any VPC related queries, the VPC pages would have incorrect information unless we refreshed the page. This change invalidates VPC related queries when deleting a linode so that the vpc pages are up to date.

## Changes  🔄
List any change relevant to the reviewer.
- Added utility function to get all vpcs a linode is assigned to with, based on its configs
- Invalidate vpc-related queries for vpcs a linode is assigned to 
- Added some extra spacing underneath the 'Reboot linode banner' on a VPC details page -- seems like there was spacing missing, added more to make it align more with the mockups 
- updated VPC todo comments to be consistent - `//@TODO VPC: ...todos...`

## Preview 📷
Before:
https://github.com/linode/manager/assets/139280159/eb4fae00-dbf0-49af-b54f-bb3ec449e307

After:
https://github.com/linode/manager/assets/139280159/24012745-b602-4aef-abd6-abe724c85c3c

### Prerequisites
(How to setup test environment)
Use the dev environment, with VPC tags, 
- Have a linode and at least two vpc's created (each with at least one subnet)

### Verification steps 
(How to verify changes)
Verify that the VPC landing page and specific VPC detail pages are correctly updated after you delete a linode that is assigned to a VPC. Test the following cases:
- Linode assigned to one vpc one subnet
- Linode assigned to one vpc (multiple subnets within that VPC)
- Linode assigned to two+ VPCs

### Tests
- `yarn test LinodesLanding/utils.test.ts`

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [x] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [x] 🔐 Removing all sensitive information from the code and PR description
- [x] 🚩 Using a feature flag to protect the release
- [x] 👣 Providing comprehensive reproduction steps
- [x] 📑 Providing or updating our documentation
- [x] 🕛 Scheduling a pair reviewing session
- [x] 📱 Providing mobile support
- [x] ♿  Providing accessibility support

### CODE REVIEW

**Review by abailly-akamai — APPROVED** [review]
Query invalidated and approach is solid to me ✅

File: packages/manager/src/features/Linodes/LinodesLanding/DeleteLinodeDialog.tsx [hunk 1]
```diff
@@ -17,13 +26,28 @@ interface Props {
...
+  );
+
   const { linodeId, onClose, onSuccess, open } = props;
 
   const { data: linode } = useLinodeQuery(
     linodeId ?? -1,
     linodeId !== undefined && open
   );
 
+  const { data: configs } = useAllLinodeConfigsQuery(
```
[the remark this answers is not in the document]
Comment (bnussman-akamai) [reply packages/manager/src/features/Linodes/LinodesLanding/DeleteLinodeDialog.tsx:46]: Yeah, you are correct! This is good!

File: packages/manager/src/features/Linodes/LinodesLanding/DeleteLinodeDialog.tsx [hunk 2]
```diff
@@ -36,6 +60,20 @@ export const DeleteLinodeDialog = (props: Props) => {
...
+      vpcIds.forEach((vpcId) => {
+        queryClient.invalidateQueries([vpcQueryKey, 'vpc', vpcId]);
+        queryClient.invalidateQueries([
+          vpcQueryKey,
+          'vpc',
+          vpcId,
+          subnetQueryKey,
+        ]);
+      });
+    }
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Linodes/LinodesLanding/DeleteLinodeDialog.tsx:76]: This logic looks good to me. Generally, I prefer to have invalidation logic inside of the `useDeleteLinodeMutation`'s `onSuccess` in the `packages/manager/src/queries/linodes/linodes.ts` file but in situations like this, I think this is a fair approach. 

The one flaw of this approach is that it is technically possible (but probably unlikely) for the Linode configs to not have loaded by the time we run this logic. 

I know I told you that for invalidations, "it’s best to be as specific as possible", but I just looked at `packages/manager/src/queries/linodes/events.ts` and it looks like we handle volumes and firewalls there. Maybe we should just follow that pattern for now? The benefit of invalidating based on the event is that invalidation can happen across many instances of cloud manager that the user may have open.

Comment (dwiley-akamai) [reply packages/manager/src/features/Linodes/LinodesLanding/DeleteLinodeDialog.tsx:76]: We could go either way with it, but I like this more targeted invalidation approach

**Review by dwiley-akamai — APPROVED** [review]
VPC pages are correctly updated after Linode deletion for each of the test cases ✅
Code review ✅

**Review by bnussman-akamai — APPROVED** [review]

**Comment by coliu-akamai** [comment]
Merging as cypress test failures are not due to vpc issues/not from this PR!

**Pull request merged**